code/HabitTracker/app/src/main/java/com/example/habittracker/utils/CustomDatePicker.java

```diff
@@ -36,16 +46,35 @@ public void onClick(View v) {
         });
     }
 
+    /**
+     * override onDateSet function to set year,month,day for the calendar object.
+     * @param view
+     * @param year
+     * @param month
+     * @param dayOfMonth
+     */
     @Override
     public void onDateSet(android.widget.DatePicker view, int year, int month, int dayOfMonth) {
         myCalendar.set(Calendar.YEAR, year);
         myCalendar.set(Calendar.MONTH, month);
         myCalendar.set(Calendar.DAY_OF_MONTH, dayOfMonth);
         updateLabel();
     }
+
+    /**
+     * convert date object to a string with format 'YYYY-MM-dd'
+     */
     private void updateLabel() {
         String myFormat = "yyyy-MM-dd";
         SimpleDateFormat sdf = new SimpleDateFormat(myFormat, Locale.US);
         edittext.setText(sdf.format(myCalendar.getTime()));
     }
+
+    /**
+     * get the calendar object
+     * @return Calendar
+     */
+    public Calendar getDate() {
```

Comment:
I actually implemented a similar function which returns the currently selected date. (Not the Calendar) Might be worth deciding on one of them for later (or for now, depending on urgency)
``` 
  /**
     * Returns the Date the datePicker was set to.
     * @return setDate, the Date
     *         null,    if no date was selected
     */
    public Date getSetDate() {
        String myFormat = "yyyy-MM-dd";
        SimpleDateFormat sdf = new SimpleDateFormat(myFormat, Locale.US);

        Date setDate = null;
        try {
            setDate = sdf.parse(edittext.getText().toString());
        } catch (ParseException e) {
            e.printStackTrace();
        }
        return ( setDate );
    }
```

Reply:
lets keep it for now